stan/math/prim/meta/require_generics.hpp

```diff
@@ -102,6 +102,13 @@ template <template <class...> class TypeCheck, class... Check>
 struct is_eigen_value_check
     : container_value_type_check_base<is_eigen, TypeCheck, Check...> {};
 
+/**
+ * Meta type trait to check if type inherits from EigenBase.
```

Comment:
These should have their template parameters documented as well as what kind of value they're going to produce.   Same for all the rest of these traits classes.